Assets/HoloToolkit/Utilities/Scripts/Solvers/SolverRadialView.cs

```diff
@@ -0,0 +1,224 @@
+		public bool IgnoreAngleClamp = false;
+		[Tooltip("Option to ignore distance clamping")]
+		public bool IgnoreDistanceClamp = false;
+
+		[Tooltip("If true, element will orient to ReferenceDirection, otherwise it will orient to ref pos (the head is the only option currently)")]
+		public bool OrientToRefDir = false;
+        #endregion
+
+        /// <summary>
+        ///   ReferenceDirectoin is the direction of the cone.  Position to the view direction, or the movement direction
+        /// </summary>
+        /// <returns>Vector3, the forward direction to use for positioning</returns>
+        private Vector3 GetReferenceDirection()
+		{
+			Vector3 ret = Vector3.one;
+			if (ReferenceDirection == ReferenceDirectionEnum.HeadMoveDirection && solverHandler.TrackedObjectToReference == SolverHandler.TrackedObjectToReferenceEnum.Head)
+			{
+                ret = Camera.main.GetComponent<InputModule.CameraMotionInfo>().MoveDirection;
+			}
+			else
+			{
+                ret = base.solverHandler.TransformTarget != null ? base.solverHandler.TransformTarget.forward : Vector3.forward;
+            }
+            return ret;
+        }
+
+		/// <summary>
+		///   Cone may roll with head, or not.
+		/// </summary>
+		/// <returns>Vector3, the up direction to use for orientation</returns>
+		private Vector3 GetReferenceUp()
+		{
+			Vector3 ret = Vector3.up;
+			if (ReferenceDirection == ReferenceDirectionEnum.ObjectOriented)
+			{
+				ret = base.solverHandler.TransformTarget != null ? base.solverHandler.TransformTarget.up : Vector3.up;
+			}
+			return ret;
+		}
+
+		private Vector3 GetReferencePoint()
+		{
+			return base.solverHandler.TransformTarget != null ? base.solverHandler.TransformTarget.position : Vector3.zero;
+		}
+
+        /// <summary>
+        /// Solver update function used to orient to the user
+        /// </summary>
+        public override void SolverUpdate()
+        {
+
+            Vector3 desiredPos = this.WorkingPos;
+
+			if (IgnoreAngleClamp)
+			{
+				if (IgnoreDistanceClamp)
+				{
+					//desiredPos = transform.position;
+				}
+				else
+				{
+					GetDesiredOrientation_DistanceOnly(ref desiredPos);
+				}
+			}
+			else
+			{
+				GetDesiredOrientation(ref desiredPos);
+			}
+
+			// Element orientation
+			Vector3 refDirUp = GetReferenceUp();
+			Quaternion desiredRot = Quaternion.identity;
+
+            if (OrientToRefDir)
+			{
+				desiredRot = Quaternion.LookRotation(GetReferenceDirection(), refDirUp);
+			}
+			else
+			{
+				Vector3 refPoint = GetReferencePoint();
+				desiredRot = Quaternion.LookRotation(desiredPos - refPoint, refDirUp);
+			}
+
+            // If gravity aligned then zero out the x and z eulers on the rotation
+            if(ReferenceDirection == ReferenceDirectionEnum.GravityAligned)
+            {
+                desiredRot.x = desiredRot.z = 0f;
+            }
+
+            this.GoalPosition = desiredPos;
+			this.GoalRotation = desiredRot;
```

Comment:
NIT: There are strange one space off formatting issues in this file.

Reply:
I've found this in a few places, it looks completely fine in VS. I'll do a reformat.